### ISSUE TITLE
serial.serialutil.SerialException: Cannot configure port, something went wrong. Original message: PermissionError(13, 'The device connected to the system is not working.', None, 31)

### ISSUE THREAD

**Issue by josalggui** [PICC-Group/pynanovna#7]
Hello,

I’ve been using Python code to control my NanoVNA and recently came across the pynanovna package. I’ve successfully controlled the NanoVNA using both my own code and your package (with minor modification) on Ubuntu and Windows 10. However, I’ve encountered an issue when trying to use it on Windows 11.

In the __init__ method of the NanoVNAWorker class, an error occurs when attempting to access the NanoVNA. The serial module throws the following exception:

serial.serialutil.SerialException: Cannot configure port, something went wrong. Original message: PermissionError(13, 'The device connected to the system is not working.', None, 31)

I’m unsure if others have experienced this same problem or if any debugging has already been done on this issue. Any insights would be greatly appreciated.

**Comment by OdoctorG** [comment]
Is this fixed with #6. I don't experience this with my windows machine, could it be some permission issue from some other program blocking the port?

**Comment by josalggui** [comment]
No, #6 do not fix the issue. With #6 I can do connection from my ubuntu or windows 10 without errors. But windows 11 still give me the error. I will continue exploring this issue, but to luck for the moment... May be related to the firmware version? Is there a way I can know the firmware version on the nanoVNA?

**Comment by OdoctorG** [comment]
You can't do it within pynanovna, but if you download NanoVNA-App (https://nanovna.com/?page_id=141) you can upload new firmware and I think you can check your current firmware version.

**Comment by tbergkvist** [comment]
There will be a new major version of pynanovna released soon, some of the known problems will be fixed. 
However, this seems like a problem with your device and not the code itself. We will however try to include a way to get info about the firmware version in the next update.

**Comment by tbergkvist** [comment]
Good to hear! Closing this issue then.
